### DISCUSSION TITLE
Unknown layer VV while reading tech file

### DISCUSSION THREAD

**Discussion by msingh9** [The-OpenROAD-Project/OpenROAD#4249] [Q&A]
I am trying to use a proprietary pdk and I hit issue when running command pin_access.

# Global routing
pin_access -bottom_routing_layer $min_routing_layer \
           -top_routing_layer $max_routing_layer

From the error message, it looks like layer "VV" is not recognized. However, while reading tech file (earlier in the flow), It did find this layer. There are lots of warnings during tech file read but I see similar warnings for other layers too.

This is the error message I get:
[WARNING DRT-0124] Via VV with unused layer VV_XX_450_450_450_450_VV will be ignored.
[ERROR DRT-0129] Unknown layer VV for viarule VV_VIAGEN.

These are some of information and warnings while reading tech file.
[WARNING ODB-0279] parse mismatch in layer property LEF58_TYPE for layer II_QM_BOT : "TYPE SPECIALCUT LAYER IA QM ;"
[INFO ODB-0388] unsupported LEF58_ANTENNAGATEPLUSDIFF property for layer VV :"ANTENNAGATEPLUSDIFF OXIDE1 5.0 ; "

My question is -
It seems to read tech file without error at the beginning of the flow. Then why is it reading tech and libs again with the command pin_access. First time it didn't error out, but it generates error during this step.

# Global routing
pin_access -bottom_routing_layer $min_routing_layer \
           -top_routing_layer $max_routing_layer
[INFO DRT-0149] Reading tech and libs.

I have checked that layer VV is defined in the tech file.

Any suggestions and ideas on what I can try to bypass this error.

I am only interested in rough timing and area results after global routing.

**Comment by maliberty** [comment]
If you only want rough results you can skip the pin_access command.  The LEF was read earlier but the interpretation is being made now and some issue is being flagged.  The exact issue is hard to say from this info.

**Comment by msingh9** [comment]
Thanks for suggestion. This enabled me to get past this point and finish global route.
